src/components/Shared/Navbar/Search.tsx

```diff
@@ -96,8 +98,14 @@ const Search: FC<Props> = ({ hideDropdown = false }) => {
               <>
                 {profiles.map((profile: Profile) => (
                   <div key={profile?.handle} className="py-2 px-4 hover:bg-gray-100 dark:hover:bg-gray-800">
-                    <Link href={`/u/${profile?.handle}`} onClick={() => setSearchText('')}>
-                      <UserProfile profile={profile} />
+                    <Link
+                      href={isParentMessage ? `` : `/u/${profile?.handle}`}
+                      onClick={() => {
+                        onProfileSelected && onProfileSelected(profile);
```

Comment:
Would it be possible to move the previous profile link to an onProfileSelected callback so we don't have competing `href` and `onClick`s and they can both use `onClick()`?